services/director/src/simcore_service_director/producer.py

```diff
@@ -63,6 +63,44 @@ def _check_setting_correctness(setting: Dict) -> None:
         raise exceptions.DirectorException("Invalid setting in %s" % setting)
 
 
+def _parse_mount_settings(settings: List[Dict]) -> List[Dict]:
+    mounts = list()
+    for s in settings:
+        log.debug("Retrieved mount settings %s", s)
+        mount = dict()
+        mount["ReadOnly"] = True
+        if "ReadOnly" in s and s["ReadOnly"] in ["false", "False", False]:
+            mount["ReadOnly"] = False
+
+        for field in ["Source", "Target", "Type"]:
+            if field in s:
+                mount[field] = s[field]
+            else:
+                log.warning(
+                    "Mount settings have wrong format. Required keys [Source, Target, Type]"
+                )
```

Comment:
should it still append the mount in that case?

Reply:
No, because it will fail.